farm/modeling/prediction_head.py

```diff
@@ -1373,14 +1423,16 @@ def reduce_preds(self, preds):
                                                         offset_unit="token",
                                                         aggregation_level="passage",
                                                         passage_id=str(sample_idx),
-                                                        n_passages_in_doc=n_samples)
+                                                        n_passages_in_doc=n_samples,
+                                                        confidence=qa_candidate.confidence)
                                             )
 
         # TODO add switch for more variation in answers, e.g. if varied_ans then never return overlapping answers
         pos_answer_dedup = self.deduplicate(pos_answers_flat)
 
         # This is how much no_ans_boost needs to change to turn a no_answer to a positive answer (or vice versa)
         no_ans_gap = -min([nas - pbs for nas, pbs in zip(no_answer_scores, passage_best_score)])
+        no_ans_gap_confidence = -min([nas - pbs for nas, pbs in zip(no_answer_confidences, passage_best_confidence)])
```

Comment:
I thought we would leave confidence as is and not adjust it in the complicated ways we do for the "normal" score

Reply:
Since I noticed that I did all my experiments and tests with this complicated method, I prefer to us this one. It works fine and has the advantage that the confidence is consistent with the score also for no_answers.